plugin-build/src/test/kotlin/io/sentry/android/gradle/SentryPluginTest.kt

```diff
@@ -0,0 +1,195 @@
+import kotlin.test.assertTrue
+
+@Suppress("FunctionName")
+@RunWith(Parameterized::class)
+class SentryPluginTest(
+    private val androidGradlePluginVersion: String,
+    private val gradleVersion: String,
+) {
+    @get:Rule
+    val testProjectDir = TemporaryFolder()
+
+    val projectTemplateFolder = File("src/test/resources/testFixtures/appTestProject")
+
+    private lateinit var rootBuildFile: File
+    private lateinit var appBuildFile: File
+    private lateinit var runner: GradleRunner
+
+    @Before
+    fun setup() {
+
+        projectTemplateFolder.copyRecursively(testProjectDir.root)
+
+        val pluginClasspath = PluginUnderTestMetadataReading.readImplementationClasspath()
+            .joinToString(separator = ", ") { "\"$it\"" }
+
+        appBuildFile = File(testProjectDir.root, "app/build.gradle")
+        rootBuildFile = testProjectDir.writeFile("build.gradle") {
+            // language=Groovy
+            """
+            buildscript {
+              repositories {
+                google()
+                gradlePluginPortal()
+              }
+              dependencies {
+                classpath 'com.android.tools.build:gradle:$androidGradlePluginVersion'
+                // This is needed to populate the plugin classpath instead of using
+                // withPluginClasspath on the Gradle Runner.
+                classpath files($pluginClasspath)
+              }
+            }
+            
+            allprojects {
+              repositories {
+                google()
+                mavenCentral()
+                jcenter()
+              }
+            }
+            subprojects {
+              pluginManager.withPlugin('com.android.application') {
+                android {
+                  compileSdkVersion 30
+                  defaultConfig {
+                    applicationId "com.example"
+                    minSdkVersion 21
+                  }
+                  buildTypes {
+                    release {
+                      minifyEnabled true
+                      proguardFiles("src/release/proguard-rules.pro")
+                    }
+                  }
+                }
+              }
+            }
+            """.trimIndent()
+        }
+
+        runner = GradleRunner.create()
+            .withProjectDir(testProjectDir.root)
+            .withArguments("--stacktrace")
+            .withPluginClasspath()
+            .withGradleVersion(gradleVersion)
+    }
+
+    @Test
+    fun `plugin can be applied`() {
+        appBuildFile.writeText(
+            // language=Groovy
+            """
+                plugins {
+                  id "com.android.application"
+                  id "io.sentry.android.gradle"
+                }
+                
+                sentry {
+                  autoUpload = false
+                }
+            """.trimIndent()
+        )
+
+        runner.build()
+    }
+
+    @Test
+    fun `generates proguard files when enabled`() {
```

Comment:
I'd remove this test because `autoProguardConfig` is gonna be deprecated/demoved